### ISSUE TITLE
Legacy source with only `deb-src` line incorrectly shows as "Enabled"

### ISSUE THREAD

**Issue by jacobgkau** [pop-os/repoman#105]
Discovered this while testing https://github.com/pop-os/repolib/pull/50. Sort of exposed by https://github.com/pop-os/repolib/pull/47.

Sources have two on/off switches, `Source Code` and `Enabled`. The `Enabled` option controls the `deb` line for a source, and the `Source Code` option controls the corresponding `deb-src` line.

If I create a `.list` file with only a `deb-src` line, then opening that source in Repoman incorrectly says that it's `Enabled`:

![Screenshot from 2022-10-06 16-39-55](https://user-images.githubusercontent.com/7199422/194431603-67c712a3-57f3-4d15-b266-4a8dfb733398.png)

Saving in this state, without touching any of the settings, adds an enabled `deb` line:

![Screenshot from 2022-10-06 16-40-21](https://user-images.githubusercontent.com/7199422/194431623-912705ce-dddc-4137-8cf7-e12283bd8dc8.png)

What should happen instead is that opening the repo in Repoman should show `Source Code` switched on but `Enabled` switched off. This is what Repoman shows if both lines are present but the `deb` line (and only the `deb` line) is commented out:

![Screenshot from 2022-10-06 16-40-43](https://user-images.githubusercontent.com/7199422/194431642-3d4ba217-a1bf-4b7c-8710-687ee8cab616.png)

If the `deb` line is going to be automatically added, it should be commented out if it was previously absent altogether.

**Comment by isantop** [comment]
I think the correct UX in this case is to have the source switch enabled (to indicate it's source repo) and grayed out to prevent a deb repo from being created.

Are there any cases where a source code repository wouldn't be accompanied by a matching binary repository? Since `deb-src` repos provide access to source packages that themselves build corresponding binary packages, I can't think of any case where it's desirable to have `deb-src` without a matching `deb` repo.

**Comment by jacobgkau** [comment]
> Are there any cases where a source code repository wouldn't be accompanied by a matching binary repository? Since `deb-src` repos provide access to source packages that themselves build corresponding binary packages,

Based on that description of `deb-src` repos, I could imagine a case where someone wants to build a package themselves and does not want the binary repo if they don't intend to install a prebuilt binary. I'm not sure if that's actually a possible use of `deb-src`, but I don't see why it couldn't be for a repo containing a small set of packages.

(IMO, the issue is more about the idempotence aspect of Repoman being able to accurately display the state of the config file and not change the effective configuration if the user doesn't change anything in the GUI, so I would argue the use case is somewhat irrelevant.)

> I think the correct UX in this case is to have the source switch enabled (to indicate it's source repo) and grayed out to prevent a deb repo from being created.

I agree on the source switch being enabled (as is currently the case), but I don't understand what you mean about anything being greyed out. Why wouldn't we want to keep the Enabled switch toggleable (just disabled by default if there's no `deb` line) so someone can enable the binary repo if they want to?

If the issue is the wording of "enabled" (i.e. we don't want to imply that the source-only repo is not enabled with a disabled `Enabled` switch), then perhaps a better UX path would be to clarify the `Enabled` toggle's label to make it more clear it's only talking about the binary package repo and not the source repo.

**Comment by isantop** [comment]
Sorry, the UX I had in mind was that the "Enabled" Switch would continue to function as normal (enable or disable the switch) and the "Source Code" switch would be switched "on" to indicate that it was enabled and disabled/insensitive to prevent changes, since there isn't a clear use case where having that switch would make sense.

**Comment by isantop** [comment]
A screenshot indicating the desired final state: 
![image](https://user-images.githubusercontent.com/5883565/194944694-546d3af1-c526-4a77-a9d7-fce259f0f1fa.png)

**Comment by isantop** [comment]
Probably not a bad idea to add a tooltip to explain why that switch isn't toggleable

**Comment by jacobgkau** [comment]
I don't think this makes sense. This turns the Source Code switch from a toggle into just an indicator.

> since there isn't a clear use case where having that switch would make sense.

What if someone has only the `deb-src` line added and wants to enable the regular `deb` line? This is already a possible configuration with the current GUI. This issue is just a bug with what's shown/read into the GUI if the `deb` line is missing/removed instead of just commented out. I think trying to handle it as a separate state in the GUI like this is just overcomplicating things.

**Comment by isantop** [comment]
The issue is in creating a consistent experience. 

With the current situation, the label of the switch doesn't match the action taken, as it enables the binary repository. 

We can change the label of the of the switch, but then if they enable it then the next time they load it, they will end up with a different switch without the ability to disable the binary repository independently of the source-code repo, making it difficult to revert back to the earlier configuration. 

My thoughts are that since Repolib/Repoman will never create that situation, the only way a user will end up in that configuration is for them to install a software package which adds it or by manually creating a source like that. In the first case, we likely don't want to make it easy to potentially break the configuration since it's likely to be automatically configured, and in the second case simply informing the user about the state should be sufficient because they have likely demonstrated sufficient base knowledge to manually change it. I'd rather err on the side of not making irreversible/confusing changes, especially in a user-facing GUI. 

One possibility is simply separating out source and binary repos into separate listings in the repository list, but I feel that one of the nicest features of Repoman post-repolib is the clutter-free repository listing. 

The other possible option is adding a switch for the Binary repo as well, but as we discussed earlier there isn't a large use case for disabling the binary repo while leaving the source code repo enabled. That's generally going to be an advanced option that is better left to manual configuration anyway. I'm more comfortable communicating the actual state and preventing GUI changes when presented with a rare situation to prevent possible breakage. 

One note: The switch would remain a toggle for repos which already have a binary repository enabled, it would only be disabled if there isn't a corresponding binary repo.

**Comment by jacobgkau** [comment]
> The issue is in creating a consistent experience.

Having the `Source Code` switch active if `deb` is commented out but inactive if `deb` doesn't exist is not consistent.

> We can change the label of the of the switch, but then if they enable it then the next time they load it, they will end up with a different switch without the ability to disable the binary repository independently of the source-code repo, making it difficult to revert back to the earlier configuration.

What? If the label of the `Enabled` switch was just changed to something like `Binary Packages`, then any combination of binary/source enabled/disabled would be possible. At that point, just fix the bug so `Binary Packages` doesn't show as enabled when opening a source that's missing its `deb` line.

> the only way a user will end up in that configuration is for them to install a software package which adds it or by manually creating a source like that. In the first case, we likely don't want to make it easy to potentially break the configuration since it's likely to be automatically configured,

This is a lot of speculation and assumption about a theoretical use case. This is why I think it would be better to make the interface straightforward and able to handle any valid text file configuration (consistently) rather than trying to guess what people or applications "want" to be possible.
